resources/manager/data.go

```diff
@@ -62,77 +65,95 @@ func (d *Data) Output() <-chan ResourceMap {
 
 // Run updates the cache using Fetcher implementations.
 func (d *Data) Run(ctx context.Context) error {
-	updates := make(chan update)
-
-	var wg sync.WaitGroup
-	d.wg = &wg
-
-	for _, key := range d.fetchers.Keys() {
-		wg.Add(1)
-		go func(k string) {
-			defer wg.Done()
-			d.fetchWorker(ctx, updates, k)
-		}(key)
-	}
-
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		d.fetchManager(ctx, updates)
-	}()
-
+	go d.fetchAndSleep(ctx)
 	return nil
 }
 
-func (d *Data) fetchWorker(ctx context.Context, updates chan update, k string) {
+func (d *Data) fetchAndSleep(ctx context.Context) {
+	// Happens once in a lifetime of cloudbeat and then enters the loop
+	d.fetchIteration(ctx)
 	for {
 		select {
+		case <-d.stop:
+			logp.L().Errorf("fetchers manager stopped")
+			return
 		case <-ctx.Done():
+			logp.L().Errorf("fetchers manager canceled")
 			return
-		default:
-			if !d.fetchers.ShouldRun(k) {
-				break
-			}
+		case <-time.After(d.interval):
+			d.fetchIteration(ctx)
+		}
+	}
+}
+
+func (d *Data) fetchIteration(ctx context.Context) {
+	logp.L().Infof("manager trigger fetching using %d fetchers", len(d.fetchers.Keys()))
+	d.wg = &sync.WaitGroup{}
```

Comment:
kind of dangerous if this function is ever called in parallel.

Reply:
You are right, I added a comment.
You think there is something else should be done here?

Reply:
Since this will change many times more a comment is sufficient for now.